evaluation/evaluate.py

```
@@ -119,6 +122,15 @@ def get_results_dir(self, output_dir: Path) -> Path:
 
         dir_name = "__".join(filter(None, components))  # Filter None/empty strings
         config_dir = output_dir / dir_name
+
+        # Make sure the directory does not exist, if it does, add a number to the end
+        # This is to avoid overwriting results
+        if config_dir.exists():
```

should we raise an error instead?

---

The previous behavior was to overwrite the results. I realized that even if the configuration directory already exists, the press initialization might be different. For example: 
- ExpectedAttention with:  n_sink=4, Llama3.1-8B, Ruler 4096 
- ExpectedAttention with:  n_sink=10, Llama3.1-8B, Ruler 4096 
would create the same directory name but different config.yaml files. The reason is that the name of the directory only depends on model dataset and press, not on the press parameters (else it would be too long).

---

ok makes sense, then leave it as is